src/SDK.ts

```diff
@@ -795,7 +845,11 @@ export default class SDK implements ISDK {
     return new Promise(async (resolve, reject) => {
       try {
         const response = await axiosInstance(options);
-        const { data } = response;
+        const { data, headers } = response;
+        if (headers?.authcodenonce) {
+          this.configuration.authCodeNonce = headers?.authcodenonce;
+        }
```

Comment:
my suggestion is to abstract this to a setAtuhCodeNonce(headers) function , instead of repeat it 10 times ,

Reply:
done